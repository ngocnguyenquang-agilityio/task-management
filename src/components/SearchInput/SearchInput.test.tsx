import { render, screen, fireEvent } from '@testing-library/react';

// Components
import { SearchInput } from './';

describe('SearchInput component', () => {
  const handleChange = jest.fn();

  test('renders the default SearchInput', () => {
    expect(render(<SearchInput onChange={handleChange} />)).toMatchSnapshot();
  });

  test('Test handles change events', () => {
    render(<SearchInput onChange={handleChange} />);
    fireEvent.change(screen.getByRole('textbox'), {
      target: { value: 'Test input' },
    });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  test('Test when the disabled prop is set', () => {
    render(<SearchInput disabled onChange={handleChange} />);
    expect(screen.getByRole('textbox')).toBeDisabled();
  });
});
