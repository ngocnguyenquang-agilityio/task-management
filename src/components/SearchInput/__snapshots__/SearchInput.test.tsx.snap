// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component renders the default SearchInput 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative w-full"
      >
        <div
          class="pointer-events-none absolute inset-y-0 start-0 flex items-center ps-3"
        >
          <svg
            class="h-6 w-6 text-slate-500"
            fill="currentColor"
            height="1em"
            id="search"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 24 24"
            width="1em"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g>
              <path
                d="M20.031,20.79c0.46,0.46,1.17-0.25,0.71-0.7l-3.75-3.76c1.27-1.41,2.04-3.27,2.04-5.31
		c0-4.39-3.57-7.96-7.96-7.96s-7.96,3.57-7.96,7.96c0,4.39,3.57,7.96,7.96,7.96c1.98,0,3.81-0.73,5.21-1.94L20.031,20.79z
		 M4.11,11.02c0-3.84,3.13-6.96,6.96-6.96c3.84,0,6.96,3.12,6.96,6.96c0,3.84-3.12,6.96-6.96,6.96C7.24,17.98,4.11,14.86,4.11,11.02
		z"
              />
            </g>
          </svg>
        </div>
        <input
          class="p-4 border border-zinc-300 focus:outline focus:outline-zinc-300 rounded-lg w-full dark:border-gray-700 dark:focus:outline-gray-700 dark:focus:border-gray-700 disabled:cursor-not-allowed disabled:opacity-50 bg-transparent text-zinc-500 dark:text-white ps-10"
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative w-full"
    >
      <div
        class="pointer-events-none absolute inset-y-0 start-0 flex items-center ps-3"
      >
        <svg
          class="h-6 w-6 text-slate-500"
          fill="currentColor"
          height="1em"
          id="search"
          stroke="currentColor"
          stroke-width="0"
          version="1.1"
          viewBox="0 0 24 24"
          width="1em"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <g>
            <path
              d="M20.031,20.79c0.46,0.46,1.17-0.25,0.71-0.7l-3.75-3.76c1.27-1.41,2.04-3.27,2.04-5.31
		c0-4.39-3.57-7.96-7.96-7.96s-7.96,3.57-7.96,7.96c0,4.39,3.57,7.96,7.96,7.96c1.98,0,3.81-0.73,5.21-1.94L20.031,20.79z
		 M4.11,11.02c0-3.84,3.13-6.96,6.96-6.96c3.84,0,6.96,3.12,6.96,6.96c0,3.84-3.12,6.96-6.96,6.96C7.24,17.98,4.11,14.86,4.11,11.02
		z"
            />
          </g>
        </svg>
      </div>
      <input
        class="p-4 border border-zinc-300 focus:outline focus:outline-zinc-300 rounded-lg w-full dark:border-gray-700 dark:focus:outline-gray-700 dark:focus:border-gray-700 disabled:cursor-not-allowed disabled:opacity-50 bg-transparent text-zinc-500 dark:text-white ps-10"
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
